compute/data_spark_version.go

```diff
@@ -0,0 +1,94 @@
+package compute
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceClusterSparkVersions() *schema.Resource {
+	return &schema.Resource{
+		Read: DataSourceClusterSparkVersionsRead,
```

Comment:
I believe readcontext should be available for datasources as well. We should switch to the functions that return diags.